service/src/main/resources/templates/index.html

```diff
@@ -110,7 +85,48 @@
     };
   }
 
-  window.onload = function() {
+    // Examples cache
+    const examplesCache = {};
+
+    // Custom plugin for logic that happens before a parameters element is created, hardcoded
+    // to work with the dataset example in the PUT dataset operation.
+    const LoadExternalValuePlugin = () => {
+      return {
+        wrapComponents: {
+          // wrapping https://github.com/swagger-api/swagger-ui/blob/master/src/core/components/parameters/parameters.jsx
+          parameters: (Original, { React }) => (props) => {
+            // These hardcoded values should be replaced with code that determines the values
+            // using the user's current selection.
+            const contentType = 'application/json'
+            const activeExamplesKey = 'dataset'
+            const operation = props.operation
+            const externalValue = operation.getIn(['requestBody', 'content', contentType, 'examples', activeExamplesKey, 'externalValue'])
+
+            if (externalValue) {
+              if (examplesCache[externalValue]) {
+                // If the examples cache contains the externalValue key, set the example value
+                // directly from the cache.
+                props.operation = operation.setIn(['requestBody', 'content', contentType, 'examples', activeExamplesKey, 'value'], examplesCache[externalValue])
+              } else {
+                // Fetch the external file.
+                fetch(externalValue)
```

Comment:
With fresh eyes I think I am starting to see why this isn't refreshing, but will need to play with it a little more